### PR TITLE
Minor improvements in spring-test

### PR DESCRIPTION
**Pull request by imkulwant**
## Minor fixes in spring-test

Commits on Nov 2, 2022
[Remove final modifier from private method](https://github.com/spring-projects/spring-framework/pull/29413/commits/10bd12a6b194159f38332b70dcbba583437bd9a8)
 
[Use Standard Charset object](https://github.com/spring-projects/spring-framework/pull/29413/commits/598f0d8a3af6e7f6b33e864bde6a45f8d4aabdf4)
 
[Remove unnecessary toString call](https://github.com/spring-projects/spring-framework/pull/29413/commits/669ad50745a75250972d30da86e336c3b75bd5c8)

[Remove unnecessary static modifier](https://github.com/spring-projects/spring-framework/pull/29413/commits/860a271cc006ace65811aa2be6a949753057787b)

[Refactor to use enhanced switch](https://github.com/spring-projects/spring-framework/pull/29413/commits/412a253aba440ee2b13ba46515e48a91f176bd34)

[Replace concatenated strings with text blocks](https://github.com/spring-projects/spring-framework/pull/29413/commits/9f65f3d658ed5782b918b6179ef30f63bd23941d)

[Rely on auto-boxing where appropriate](https://github.com/spring-projects/spring-framework/pull/29413/commits/6d307c86d42db48ec911d96b3e44265097951d37)

[Remove Arrays.asList call with too few arguments](https://github.com/spring-projects/spring-framework/pull/29413/commits/961c9835a34035e6bc73d702ef243a00634f59cd)

[Use equals for String comparison](https://github.com/spring-projects/spring-framework/pull/29413/commits/dbf5bd8ac44785c367439e503ce23f9aca6cba26)

[Remove redundant code](https://github.com/spring-projects/spring-framework/pull/29413/commits/d31676f7ac2a213229feb964cbf01615cc6c3585)

[Fix imports](https://github.com/spring-projects/spring-framework/pull/29413/commits/266e36093842db5421b6b1bb0175f686f0f442ab)

[Revert "Remove Arrays.asList call with too few arguments"](https://github.com/spring-projects/spring-framework/pull/29413/commits/a61850ffc282a44a17f2a8155075bc51e3c9b59f) 

[Revert "Use equals for String comparison"](https://github.com/spring-projects/spring-framework/pull/29413/commits/c632a47e2bfa9d116e33983d48c2da57798e65b4) 

[Reduce indentation](https://github.com/spring-projects/spring-framework/pull/29413/commits/acee348bc5db5721417727e16f523d3325329c22)

[Improve text blocks in tests](https://github.com/spring-projects/spring-framework/pull/29413/commits/4ce2071864df35842c86692c0ffc9fdfd63ed561)

### CODE REVIEW

File: spring-test/src/main/java/org/springframework/test/context/junit/jupiter/SpringExtension.java [hunk 1]
```diff
@@ -159,7 +159,7 @@ private void validateAutowiredConfig(ExtensionContext context) {
 								testClass.getName(), Arrays.toString(methodsWithErrors)));
 			}, String.class);
 
-		if (errorMessage != NO_AUTOWIRED_VIOLATIONS_DETECTED) {
+		if (!NO_AUTOWIRED_VIOLATIONS_DETECTED.equals(errorMessage)) {
```
Comment (sbrannen) [inline spring-test/src/main/java/org/springframework/test/context/junit/jupiter/SpringExtension.java:162]: FYI: the original code was intentional. Identity is sufficient here.

File: spring-test/src/test/java/org/springframework/test/context/TestExecutionListenersTests.java [hunk 2]
```diff
@@ -126,13 +126,13 @@ void defaultListenersMergedWithCustomListenerInserted() {
 
 	@Test
 	void nonInheritedDefaultListeners() {
-		assertRegisteredListeners(NonInheritedDefaultListenersTestCase.class, asList(QuuxTestExecutionListener.class));
+		assertRegisteredListeners(NonInheritedDefaultListenersTestCase.class, List.of(QuuxTestExecutionListener.class));
```
Comment (sbrannen) [inline spring-test/src/test/java/org/springframework/test/context/TestExecutionListenersTests.java:129]: `asList()` is used consistently within this test class. Switching to `List.of()` in a few places therefore doesn't add value. We should use one or the other but not mix both.

So let's just leave it as-is.

In light of that, please revert the changes in this class.

File: spring-test/src/main/java/org/springframework/test/context/junit/jupiter/SpringExtension.java (hunk 1 above)
Comment (sbrannen) [inline spring-test/src/main/java/org/springframework/test/context/junit/jupiter/SpringExtension.java:162]: FYI: the original code was intentional. Identity is sufficient here.

Please revert this change.

File: spring-test/src/test/java/org/springframework/test/web/client/MockRestServiceServerTests.java [hunk 3]
```diff
@@ -183,9 +183,11 @@ public void verifyWithTimeout() {
 		this.restTemplate.getForObject("/foo", Void.class);
 
 		assertThatThrownBy(() -> server1.verify(Duration.ofMillis(100))).hasMessage(
-				"Further request(s) expected leaving 1 unsatisfied expectation(s).\n" +
-						"1 request(s) executed:\n" +
-						"GET /foo, headers: [Accept:\"application/json, application/*+json\"]\n");
+				"""
```
Comment (sbrannen) [inline spring-test/src/test/java/org/springframework/test/web/client/MockRestServiceServerTests.java:186]: Please put the `"""` on the preceding line and reduce the indentation like you've done in `SimpleRequestExpectationManagerTests`.

File: spring-test/src/test/java/org/springframework/test/web/servlet/samples/client/standalone/resultmatches/XpathAssertionTests.java [hunk 4]
```diff
@@ -224,11 +224,14 @@ public class BlogFeedController {
...
-					+ "  <icon>https://www.example.com/favicon.ico</icon>\r\n"
-					+ "</feed>\r\n\r\n";
+			return """
+					<?xml version="1.0" encoding="UTF-8"?>\r
+					<feed xmlns="http://www.w3.org/2005/Atom">\r
+						<title>Test Feed</title>\r
+						<icon>https://www.example.com/favicon.ico</icon>\r
+					</feed>\r
+					\r
+					""";
```
Comment (sbrannen) [inline spring-test/src/test/java/org/springframework/test/web/servlet/samples/client/standalone/resultmatches/XpathAssertionTests.java:234]: Instead of explicitly adding `\r` to the end of each line, how about `""".replaceAll("\n", "\r\n");`?

File: spring-test/src/test/java/org/springframework/test/web/servlet/samples/standalone/resultmatchers/XpathAssertionTests.java [hunk 5]
```diff
@@ -228,11 +228,14 @@ public class BlogFeedController {
...
-					+ "  <icon>https://www.example.com/favicon.ico</icon>\r\n"
-					+ "</feed>\r\n\r\n";
+			return """
+					<?xml version="1.0" encoding="UTF-8"?>\r
+					<feed xmlns="http://www.w3.org/2005/Atom">\r
+						<title>Test Feed</title>\r
+						<icon>https://www.example.com/favicon.ico</icon>\r
+					</feed>\r
+					\r
+					""";
```
Comment (sbrannen) [inline spring-test/src/test/java/org/springframework/test/web/servlet/samples/standalone/resultmatchers/XpathAssertionTests.java:238]: See previous comment regarding `replaceAll()`.

File: spring-test/src/test/java/org/springframework/test/context/configuration/interfaces/BootstrapWithTestInterface.java [hunk 6]
```diff
@@ -34,7 +34,7 @@
 @BootstrapWith(CustomTestContextBootstrapper.class)
 interface BootstrapWithTestInterface {
 
-	static class CustomTestContextBootstrapper extends DefaultTestContextBootstrapper {
+	class CustomTestContextBootstrapper extends DefaultTestContextBootstrapper {
```
Comment (sbrannen) [inline spring-test/src/test/java/org/springframework/test/context/configuration/interfaces/BootstrapWithTestInterface.java:37]: Every single class modified in the `Removed unnecessary static modifier` commit must be `static`.

Please revert those changes.

**Review by sbrannen — CHANGES_REQUESTED** [review]
I have requested some changes.

In addition, please note that we use the present tense command form for commit message titles. For example, "Fix typo" instead of "Fixed typo". Can you please update your commit messages accordingly?

Thanks

File: spring-test/src/test/java/org/springframework/test/util/ReflectionTestUtilsTests.java [hunk 7]
```diff
@@ -46,7 +46,7 @@
  */
 class ReflectionTestUtilsTests {
 
-	private static final Float PI = Float.valueOf((float) 22 / 7);
+	private static final Float PI = (float) 22 / 7;
```
Comment (sbrannen) [inline spring-test/src/test/java/org/springframework/test/util/ReflectionTestUtilsTests.java:49]: As a side note, I wouldn't refer to these changes as "Removed unnecessary boxing".

The boxing is necessary. It's just that the compiler will perform auto-boxing for you if you don't do it yourself.

So I'd probably use "Rely on auto-boxing where appropriate" as the commit message title.

File: spring-test/src/test/java/org/springframework/test/context/configuration/interfaces/BootstrapWithTestInterface.java (hunk 6 above)
Comment (andrei-ivanov) [reply spring-test/src/test/java/org/springframework/test/context/configuration/interfaces/BootstrapWithTestInterface.java:37]: Isn't that redundant for inner classes of interfaces?

Comment (sbrannen) [reply spring-test/src/test/java/org/springframework/test/context/configuration/interfaces/BootstrapWithTestInterface.java:37]: It is indeed redundant for nested interfaces, nested enums, and nested annotations, but the same does not apply to classes.

Those static nested classes are not "inner" classes.

In fact, your changes break the build. 

I know it's confusing, but I recommend that you read up on the differences between "static nested" classes and "inner" classes.

An "inner" class is a "nested" class, but a "static nested" class is not an "inner" class.

Comment (sbrannen) [reply spring-test/src/test/java/org/springframework/test/context/configuration/interfaces/BootstrapWithTestInterface.java:37]: Oh my... I overlooked the fact that you only made those changes to nested types in test interfaces and annotations.

I apologize. You are correct. The static modifier is not required in those.

So please keep those changes in place.

File: spring-test/src/test/java/org/springframework/test/web/servlet/samples/standalone/resultmatchers/XpathAssertionTests.java (hunk 5 above)
Comment (imkulwant) [reply spring-test/src/test/java/org/springframework/test/web/servlet/samples/standalone/resultmatchers/XpathAssertionTests.java]: Fixed [bb0484b](https://github.com/spring-projects/spring-framework/pull/29413/commits/bb0484bb300d1a1f1889bd682c93ce34cf43cc99)

File: spring-test/src/test/java/org/springframework/test/web/client/MockRestServiceServerTests.java (hunk 3 above)
Comment (imkulwant) [reply spring-test/src/test/java/org/springframework/test/web/client/MockRestServiceServerTests.java:186]: Fixed [95c2813](https://github.com/spring-projects/spring-framework/pull/29413/commits/95c28135c8804676c0951ff677a313d246fa409e)

File: spring-test/src/main/java/org/springframework/test/context/junit/jupiter/SpringExtension.java (hunk 1 above)
Comment (imkulwant) [reply spring-test/src/main/java/org/springframework/test/context/junit/jupiter/SpringExtension.java]: Reverted [88d16fb](https://github.com/spring-projects/spring-framework/pull/29413/commits/88d16fb7730febd2f35e091d20674d81d30b5fa2)

File: spring-test/src/test/java/org/springframework/test/context/TestExecutionListenersTests.java (hunk 2 above)
Comment (imkulwant) [reply spring-test/src/test/java/org/springframework/test/context/TestExecutionListenersTests.java]: Reverted [60b8775](https://github.com/spring-projects/spring-framework/pull/29413/commits/60b8775a94b890e293fae083ae540adf77b34c0a)

File: spring-test/src/test/java/org/springframework/test/util/ReflectionTestUtilsTests.java (hunk 7 above)
Comment (imkulwant) [reply spring-test/src/test/java/org/springframework/test/util/ReflectionTestUtilsTests.java:49]: Changed commit message [e41e9e1](https://github.com/spring-projects/spring-framework/pull/29413/commits/e41e9e16f30a2dec91cd5cd94d78197d1bc187e3)

File: spring-test/src/test/java/org/springframework/test/web/client/MockRestServiceServerTests.java (hunk 3 above)
Comment (imkulwant) [reply spring-test/src/test/java/org/springframework/test/web/client/MockRestServiceServerTests.java:186]: Test fails if we put `"""` on preceding line. [4bc8211](https://github.com/spring-projects/spring-framework/pull/29413/commits/4bc82118c71cf794df5578c522a4ed0261e49a6d)

I think the message have a newline at the end.

**Comment by imkulwant** [comment]
> I have requested some changes.
> 
> In addition, please note that we use the present tense command form for commit message titles. For example, "Fix typo" instead of "Fixed typo". Can you please update your commit messages accordingly?
> 
> Thanks

Updated commit messages.

File: spring-test/src/test/java/org/springframework/test/web/client/MockRestServiceServerTests.java (hunk 3 above)
Comment (sbrannen) [reply spring-test/src/test/java/org/springframework/test/web/client/MockRestServiceServerTests.java:186]: Sorry, I was unclear. I meant the leading `"""`, not the trailing `"""`.

No worries. I'll take another look before merging.

**Review by sbrannen — APPROVED** [review]
